authCipher.js

```diff
@@ -17,13 +18,39 @@ function xorTest (a, b) {
   return out
 }
 
+function calcIv (self, iv, ck) {
```

Comment:
Do you have a source for where this algorithm came from?

Reply:
primarily https://csrc.nist.gov/publications/detail/sp/800-38d/final page 15

Reply:
partially verifed,  couldn't find the `[..., 2]` ref tho?